### PR TITLE
Various API Renames

### LINKED ISSUE #14975: Improve exception message for entity name validation
> Great find @HarshaNalluru!
> 
> @JoshLove-msft, you seem to be using the below error message which makes sense for Event Hubs where connection string and the entity name were passed to the same method. But in Service Bus, the connection string is passed to the client constructor where as the entity name is passed to the methods on the client.
> 
> > The path to an Service Bus entity may be specified as part of the connection string or as a separate value, but not both.
> 
> I would recommend using a different error message. In JS, am going with
> 
> > The queue or topic name provided does not match the EntityPath in the connection string passed to the ServiceBusClient constructor.
> 
> Thoughts?

Makes sense!

_Originally posted by @JoshLove-msft in https://github.com/Azure/azure-sdk-for-js/pull/11025#discussion_r485129563_

### LINKED ISSUE #14986: Rename Proxy property to WebProxy
This will ensure clarity if there ever ends up being AmqpProxy support.
https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/servicebus/Azure.Messaging.ServiceBus/src/Client/ServiceBusClientOptions.cs#L38

### PR DESCRIPTION
**Pull request by JoshLove-msft**
- SendVia -> TransactionQueueOrTopicName
- ViaPartitionKey -> TransactionPartitionKey
- Proxy -> WebProxy

Make MaxReceiveTime internal within the ProcessorOptions.
Update entity name exception message.

Fixes https://github.com/azure/azure-sdk-for-net/issues/14986, https://github.com/azure/azure-sdk-for-net/issues/14975, https://github.com/azure/azure-sdk-for-net/issues/10646

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpClient.cs [hunk 1]
```diff
@@ -99,7 +99,7 @@ internal class AmqpClient : TransportClient
                 ServiceEndpoint,
                 credential,
                 options.TransportType,
-                options.Proxy);
+                options.WebProxy);
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpClient.cs:102]: I'm not sure where this came from, but I question the drift from the naming for the same concept in Event Hubs across the languages.

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpSender.cs [hunk 2]
```diff
@@ -446,9 +446,9 @@ public override async Task CloseAsync(CancellationToken cancellationToken)
                         entry[ManagementConstants.Properties.PartitionKey] = message.PartitionKey;
                     }
 
-                    if (!string.IsNullOrWhiteSpace(message.ViaPartitionKey))
+                    if (!string.IsNullOrWhiteSpace(message.TransactionPartitionKey))
                     {
-                        entry[ManagementConstants.Properties.ViaPartitionKey] = message.ViaPartitionKey;
+                        entry[ManagementConstants.Properties.ViaPartitionKey] = message.TransactionPartitionKey;
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpSender.cs:451]: nit:  Should the constant member also be renamed to match?

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Client/ServiceBusClientOptions.cs [hunk 3]
```diff
@@ -107,7 +107,7 @@ internal void AddPlugin(ServiceBusPlugin plugin)
             new ServiceBusClientOptions
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Client/ServiceBusClientOptions.cs:107]: nit:   There's an extra set of comment slashes indented on line 103

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusSender.cs [hunk 4]
```diff
@@ -65,9 +65,9 @@ public bool IsClosed
         internal ServiceBusEventSource Logger { get; set; } = ServiceBusEventSource.Log;
 
         /// <summary>
-        /// In the case of a via-sender, the message is sent to <see cref="EntityPath"/> via <see cref="ViaEntityPath"/>; null otherwise.
+        /// In the case of a via-sender, the message is sent to <see cref="EntityPath"/> via <see cref="TransactionEntityPath"/>; null otherwise.
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusSender.cs:68]: I'm not sure that this helps to explain the concept to those that may not be familiar.   In particular, the term `via-sender` may benefit from some more words to define the concept.

**Review by jsquire — APPROVED** [review]

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpClient.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpClient.cs:102]: This was based on a comment from Clemens in the Api View.

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusSender.cs (hunk 4 above)
Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Sender/ServiceBusSender.cs:68]: linked to the options property which has the full explanation.

**Review by MiYanni — APPROVED** [review]

**Pull request merged**